File: backend/src/main/java/nutritionAPI/Product.java
package nutritionAPI;

import io.quarkus.hibernate.orm.panache.PanacheEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Index;
import jakarta.persistence.Table;

@Entity
@Table(name = "products")
public class Product extends PanacheEntity {
    @Column(length = 100)
    public String name;

    @Column(length = 100)
    public String brand;

    public int portionSize;
    public int calories;
    public double proteinsInGrams;
    public double fatsInGrams;
    public double carbsInGrams;
}

File: backend/productApi/infrastructure/entity/product.go
package entity

type Product struct {
	ID                  uint    `json:"ID" gorm:"primaryKey"`
	Name                string  `json:"Name"`
	Brand               string  `json:"Brand"`
	CaloriesPer100Grams int     `json:"CaloriesPer100Grams"`
	ProteinsInGrams     float64 `json:"ProteinsInGrams"`
	FatsInGrams         float64 `json:"FatsInGrams"`
	CarbsInGrams        float64 `json:"CarbsInGrams"`
}
